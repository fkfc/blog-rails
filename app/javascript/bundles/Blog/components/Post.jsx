import React from 'react';
import Comment from './Comment';

function Post(props) {
  const { post, comments, postPath } = props;
  const { title, content, created_at } = post;
  const editPostPath = `${postPath}/edit`;

  const renderComments = () => {
    if (typeof comments === 'undefined' || comments === null) return null;
    return comments.map( (comment, index) => {
      const key = `comment${index}`;
      return (
        <Comment key={key} comment={comment} /> 
      )
    });
  }

  return (
    <div className="card">
      <h3 className="card-header">{title}</h3>
      <div className="card-body">
        <p className="card-text">{content}</p>
        
        
        <div className="boxFooter">
          <div className='text'>
            Posted at {created_at}
          </div>          
          <div className="btn-group" role="group">
            <a href={editPostPath} className="btn btn-secondary">Edit</a>
            <a href={postPath} className="btn btn-secondary" rel="nofollow" data-method="delete" >Remove</a>
          </div>
        </div>

      </div>
    </div>
  );
}

export default Post;
