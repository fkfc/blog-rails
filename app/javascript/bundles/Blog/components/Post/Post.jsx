import React from 'react';
import PropTypes from 'prop-types';

function Post(props) {
  const {
    postPath, title, author, content, createdAt,
  } = props;
  const editPostPath = `${postPath}/edit`;
  const authorText = author ? ` by ${author}` : '';

  return (
    <div className="card">
      <h3 className="card-header">{title}</h3>
      <div className="card-body">
        <p className="card-text">{content}</p>

        <div className="boxFooter">
          <div className="text">
            Posted at &nbsp;
            {createdAt}
            {authorText}
          </div>
          <div className="btn-group" role="group">
            <a href={editPostPath} className="btn btn-secondary">Edit</a>
            <a href={postPath} className="btn btn-secondary" rel="nofollow" data-method="delete">Remove</a>
          </div>
        </div>

      </div>
    </div>
  );
}

Post.propTypes = {
  postPath: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  author: PropTypes.string,
};

Post.defaultProps = {
  author: null,
};

export default Post;
