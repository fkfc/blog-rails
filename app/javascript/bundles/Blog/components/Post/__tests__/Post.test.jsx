import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Post from '../Post';

describe('Testing the Post Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      postPath: '/postPath',
      title: 'My Title',
      content: 'My content',
      createdAt: 'Month 00th, 9999',
    };
  });
  it('Should render correctly', () => {
    const wrapper = shallow(<Post {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Contains the post title', () => {
    const wrapper = shallow(<Post {...props} />);
    expect(wrapper.text()).toContain(props.title);
  });
  it('Includes the post content text', () => {
    const wrapper = shallow(<Post {...props} />);
    expect(wrapper.text()).toContain(props.content);
  });
});
