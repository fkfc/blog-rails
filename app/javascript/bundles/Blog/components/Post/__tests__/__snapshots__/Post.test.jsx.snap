// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Post Component Should render correctly 1`] = `
<div
  className="card"
>
  <h3
    className="card-header"
  >
    My Title
  </h3>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      My content
    </p>
    <div
      className="boxFooter"
    >
      <div
        className="text"
      >
        Posted at
        Month 00th, 9999
      </div>
      <div
        className="btn-group"
        role="group"
      >
        <a
          className="btn btn-secondary"
          href="/postPath/edit"
        >
          Edit
        </a>
        <a
          className="btn btn-secondary"
          data-method="delete"
          href="/postPath"
          rel="nofollow"
        >
          Remove
        </a>
      </div>
    </div>
  </div>
</div>
`;
